common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -92,12 +92,25 @@ bool socket_thread::init()
 
 void socket_thread::set_select_timeout(unsigned msec)
 {
+    m_select_timeout_msec      = msec;
+    m_skip_next_select_timeout = false;
```

Comment:
I think it better to set this flag to `false` inside the `work()` method.
It'll make the code a little bit more readable, since the flow will look like:
```cpp
if (m_skip_next_select_timeout) {
    select.setTimeout(0);
}

select.selectSocket();

if (m_skip_next_select_timeout) {
    m_skip_next_select_timeout = false;
    select.setTimeout(m_select_timeout_msec);
}
```

Reply:
I moved the `m_skip_next_select_timeout = false;` statement outside as you requested. But I left the skipping of the timeout to the skip function. It is not that critical IMO, and with my implementation, I don't need to change the `setTimeout` function on the select class to get msec value instead of proprietary struct.